packages/cloud-cognitive/src/components/Datagrid/Datagrid/DatagridToolbar.js

```diff
@@ -9,14 +9,110 @@ import React from 'react';
 import { DataTable } from 'carbon-components-react';
 import { pkg } from '../../../settings';
 
+import { OverflowMenu } from 'carbon-components-react';
+import OverflowMenuItem from 'carbon-components-react/lib/components/OverflowMenuItem';
+import { TableBatchActions } from 'carbon-components-react';
+import { TableBatchAction } from 'carbon-components-react';
+import { Activity16 } from '@carbon/icons-react';
+
 const blockClass = `${pkg.prefix}--datagrid`;
 
 const { TableToolbar } = DataTable;
 
+const DatagridActions = (datagridState) => {
+  const {
+    selectedFlatRows,
+    setGlobalFilter,
+    RowSizeDropdown,
+    rowSizeDropdownProps,
+  } = datagridState;
+  const downloadCsv = () => {
+    alert('Downloading...');
+  };
+  const { TableToolbarContent, TableToolbarSearch } = DataTable;
+
+  const refreshColumns = () => {
+    alert('refreshing...');
+  };
+  const leftPanelClick = () => {
+    alert('open/close left panel...');
+  };
+  const searchForAColumn = 'Search';
+  const isNothingSelected = selectedFlatRows.length === 0;
+
+  // let numItemsToolbar = 5;
+
+  return (
+    isNothingSelected && (
+      <React.Fragment>
+        <OverflowMenu>
+          <OverflowMenuItem
+            itemText="Left Panel"
+            onClick={leftPanelClick}
+          ></OverflowMenuItem>
+          <OverflowMenuItem
+            itemText="Download CSV"
+            onClick={downloadCsv}
+          ></OverflowMenuItem>
+          <OverflowMenuItem
+            itemText="Refresh Columns"
+            onClick={refreshColumns}
+          ></OverflowMenuItem>
+          <RowSizeDropdown {...rowSizeDropdownProps}></RowSizeDropdown>
+        </OverflowMenu>
+        <TableToolbarContent>
+          <TableToolbarSearch
+            size="xl"
+            id="columnSearch"
+            persistent
+            placeholder={searchForAColumn}
+            onChange={(e) => setGlobalFilter(e.target.value)}
+          />
+        </TableToolbarContent>
+      </React.Fragment>
+    )
+  );
+};
+
+const DatagridBatchActions = (datagridState) => {
+  const { selectedFlatRows, toggleAllRowsSelected } = datagridState;
+  const totalSelected = selectedFlatRows && selectedFlatRows.length;
+  const onBatchAction = () => alert('Batch action');
+  const actionName = 'Action';
+  const selectAll = 'Select All';
+  // const selectAllButton = 'Select All';
+  // const selectAllButtonAction = () => alert(`Select All}`);
+
+  const selectAllButtonAction = () => {
+    toggleAllRowsSelected(true);
+    alert('Select All');
```

Comment:
Same as above.

Reply:
I have kept this in DatagridToolbar.js, because this is one of the new functionalities that we are implementing in Batch Actions.